src/macaron/repo_finder/provenance_extractor.py

```diff
@@ -355,3 +358,354 @@ def check_if_repository_purl_and_url_match(url: str, repo_purl: PackageURL) -> b
         purl_path = f"{repo_purl.namespace}/{purl_path}"
     # Note that the urllib method includes the "/" before path while the PURL method does not.
     return f"{parsed_url.hostname}{parsed_url.path}".lower() == f"{expanded_purl_type or repo_purl.type}/{purl_path}"
+
+
+class ProvenanceBuildDefinition(ABC):
```

Comment:
I wonder if these or some of these abstractions should be put within the package `src/macaron/slsa_analyzer/provenance` as they are closely related to the provenance format 🤔 ?

Reply:
I leave further refactoring to @benmss based on the changes he is making for provenance processing and verification in other PRs.